sdk/batch/Azure.ResourceManager.Batch/src/Generated/Models/BatchImageReference.cs

```diff
@@ -57,14 +57,18 @@ public BatchImageReference()
         /// <param name="sku"> For example, 18.04-LTS or 2022-datacenter. </param>
         /// <param name="version"> A value of 'latest' can be specified to select the latest version of an image. If omitted, the default is 'latest'. </param>
         /// <param name="id"> This property is mutually exclusive with other properties. The Azure Compute Gallery Image must have replicas in the same region as the Azure Batch account. For information about the firewall settings for the Batch node agent to communicate with the Batch service see https://docs.microsoft.com/en-us/azure/batch/batch-api-basics#virtual-network-vnet-and-firewall-configuration. </param>
+        /// <param name="sharedGalleryImageId"> This property is mutually exclusive with other properties and can be fetched from shared gallery image GET call. </param>
+        /// <param name="communityGalleryImageId"> This property is mutually exclusive with other properties and can be fetched from community gallery image GET call. </param>
         /// <param name="serializedAdditionalRawData"> Keeps track of any properties unknown to the library. </param>
-        internal BatchImageReference(string publisher, string offer, string sku, string version, ResourceIdentifier id, IDictionary<string, BinaryData> serializedAdditionalRawData)
+        internal BatchImageReference(string publisher, string offer, string sku, string version, ResourceIdentifier id, string sharedGalleryImageId, string communityGalleryImageId, IDictionary<string, BinaryData> serializedAdditionalRawData)
         {
             Publisher = publisher;
             Offer = offer;
             Sku = sku;
             Version = version;
             Id = id;
+            SharedGalleryImageId = sharedGalleryImageId;
```

Comment:
What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/maps/Azure.ResourceManager.Maps/src/autorest.md?plain=1#L25).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/search/Azure.Search.Documents/src/autorest.md?plain=1#L106-L115).

Reply:
same as `CommunityGalleryImageId`

Reply:
Short answer no.  Double checked with my deve and he said they are the same as CRP's definition of direct shared gallery images, which are string-typed in the form of /SharedGalleries/{GalleryName}/Images/{ImageName}(/Versions/{ImageVersion}). They don't look like a GUID or standard ARM resource identifier.

Azure.ResourceManager.Compute SDK has the same defintion in which these two properties are also defined as strings. 
 
https://github.com/Azure/azure-sdk-for-net/blob/409dc1f050f27234584ab6f0fc4a45ebc413fc33/sdk/compute/Azure.ResourceManager.Compute/src/Generated/Models/ImageReference.cs#L54C23-L54C50